src/test/java/fi/aalto/cs/apluscourses/intellij/actions/SubmitExerciseActionTest.java

```diff
@@ -0,0 +1,279 @@
+
+public class SubmitExerciseActionTest {
+
+  Course course;
+  Exercise exercise;
+  Group group;
+  List<Group> groups;
+  ExerciseGroup exerciseGroup;
+  List<ExerciseGroup> exerciseGroups;
+  String fileKey;
+  String fileName;
+  SubmittableFile file;
+  SubmissionInfo submissionInfo;
+  ExerciseDataSource exerciseDataSource;
+  Main main;
+  MainViewModel mainViewModel;
+  CourseViewModel courseViewModel;
+  ExercisesTreeViewModel exercises;
+  String moduleName;
+  Path modulePath;
+  String moduleFilePath;
+  Module module;
+  Project project;
+  Path filePath;
+  DialogHelper<ModuleSelectionViewModel> moduleSelectionDialogHelper;
+  DialogHelper<SubmissionViewModel> submissionDialogHelper;
+  Dialogs.Factory<ModuleSelectionViewModel> moduleSelectionDialogFactory;
+  Dialogs.Factory<SubmissionViewModel> submissionDialogFactory;
+  MainViewModelProvider mainVmProvider;
+  FileFinder fileFinder;
+  SubmitExerciseAction.ModuleSource moduleSource;
+  Dialogs dialogs;
+  Notifier notifier;
+  AnActionEvent event;
+  SubmitExerciseAction action;
+
+  /**
+   * Called before each test.
+   *
+   * @throws IOException Never.
+   * @throws FileDoesNotExistException Never.
+   */
+  @SuppressWarnings("unchecked")
+  @Before
+  public void setUp() throws IOException, FileDoesNotExistException {
+    course = new ModelExtensions.TestCourse("91");
+    exercise = new Exercise(12, "Test exercise");
+    group = new Group(124, Collections.singletonList("Only you"));
+    groups = Collections.singletonList(group);
+    exerciseGroup = new ExerciseGroup("Test EG", Collections.singletonList(exercise));
+    exerciseGroups = Collections.singletonList(exerciseGroup);
+    fileName = "some_file.scala";
+    fileKey = "file1";
+    file = new SubmittableFile(fileKey, fileName);
+    submissionInfo = new SubmissionInfo(1, new SubmittableFile[]{file});
+    exerciseDataSource = spy(new ModelExtensions.TestExerciseDataSource());
+    doReturn(groups).when(exerciseDataSource).getGroups(course);
+    doReturn(submissionInfo).when(exerciseDataSource).getSubmissionInfo(exercise);
+    doReturn(exerciseGroups).when(exerciseDataSource).getExerciseGroups(course);
+    main = new Main(exerciseDataSource);
+
+    mainViewModel = new MainViewModel(main);
+    courseViewModel = new CourseViewModel(course);
+    mainViewModel.courseViewModel.set(courseViewModel);
+    exercises = new ExercisesTreeViewModel(Collections.singletonList(exerciseGroup));
+    mainViewModel.exercisesViewModel.set(exercises);
+
+    exercises.getGroupViewModels().get(0).getExerciseViewModels().get(0).setSelected(true);
+
+    moduleName = "MyModule";
+    modulePath = Paths.get(moduleName);
+    moduleFilePath = modulePath.resolve("MyModule.iml").toString();
+    module = mock(Module.class);
+    doReturn(moduleFilePath).when(module).getModuleFilePath();
+    doReturn(moduleName).when(module).getName();
+
+    project = mock(Project.class);
+
+    filePath = modulePath.resolve(fileName);
```

Comment:
this is like the example of what I mean here [in the earlier comment](https://github.com/Aalto-LeTech/intellij-plugin/pull/237/files#r457858232) (can be mocked)

also, taking into account the recent features, it might actually make sense to create own TestHelper classes, that in their turn extend IJ's `Light/HeavyPlatformTests` and provide the functionality to work with our own logical entities like `Submission`, `Exercise`, `Module`, etc.